panic: interface conversion: object.NetworkReference is *object.DistributedVirtualPortgroup, not *object.Network

#### Overview of the Issue

Creating vsphere-clone of existing template made by packer vsphere-iso previously.  App panics immediately on step 1: Cloning VM ... with unexpected EOF.

#### Reproduction Steps

packer validate XXX 
packer build XXX

### Packer version

Packer v1.5.2

### Simplified Packer Buildfile

{
  "builders": [
    {
      "type": "vsphere-clone",
      "cluster": "{{ user `cluster` }}",
      "convert_to_template": "true",
      "datastore": "{{ user `datastore` }}",
      "insecure_connection": "true",
      "network": "{{ user `vsphere_network` }}",
      "username": "{{ user `vsphere_user` }}",
      "password": "{{ user `vsphere_password` }}",
      "ram": "1024",
      "resource_pool": "{{ user `resource_pool` }}",
      "vcenter_server": "{{ user `vsphere_server` }}",
      "vm_name": "{{ user `build_name` }}",
      "communicator": "ssh",
      "cpus": "1",
      "disk_size": "30720",
      "pause_before_connecting": "2m",
      "shutdown_command": "echo 'packer' | sudo -S shutdown -P now",
      "ssh_password": "{{ user `packer_password` }}",
      "ssh_username": "packer",
      "template": "{{ user `vmware_template` }}"
    }
  ],
yada, yada, yada

### Operating system and Environment details

VMWare RHEL 7.6

### Log Fragments and crash.log files

Template validated successfully.
vsphere-clone: output will be in this color.

==> vsphere-clone: Cloning VM...
Build 'vsphere-clone' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> vsphere-clone: unexpected EOF

==> Builds finished but no artifacts were created.
panic: interface conversion: object.NetworkReference is *object.DistributedVirtualPortgroup, not *object.Network
2020/02/14 10:35:43 packer.1.5.2 plugin:
2020/02/14 10:35:43 packer.1.5.2 plugin: goroutine 161 [running]:
...
2020/02/14 10:35:43 [INFO] (telemetry) ending vsphere-clone
2020/02/14 10:35:43 ui error: Build 'vsphere-clone' errored: unexpected EOF
2020/02/14 10:35:43 /usr/local/bin/packer.1.5.2: plugin process exited
2020/02/14 10:35:43 machine readable: error-count []string{"1"}
2020/02/14 10:35:43 ui error:

[crash.1.5.4.log](https://github.com/hashicorp/packer/files/4214334/crash.1.5.4.log)
Similar crash with 1.5.4 - investigating the changes.

***
==> Builds finished but no artifacts were created.
panic: interface conversion: object.NetworkReference is *object.DistributedVirtualPortgroup, not *object.Network
2020/02/17 09:47:18 packer.1.5.4 plugin:
2020/02/17 09:47:18 packer.1.5.4 plugin: goroutine 16 [running]:
***

--- a/builder/vsphere/driver/network.go
+++ b/builder/vsphere/driver/network.go
@@ -1,6 +1,8 @@
 package driver
 
 import (
+	"fmt"
+
 	"github.com/vmware/govmomi/object"
 	"github.com/vmware/govmomi/vim25/mo"
 	"github.com/vmware/govmomi/vim25/types"
@@ -8,7 +10,7 @@ import (
 
 type Network struct {
 	driver  *Driver
-	network *object.Network
+	network object.NetworkReference
 }
 
 func (d *Driver) NewNetwork(ref *types.ManagedObjectReference) *Network {
@@ -24,7 +26,7 @@ func (d *Driver) FindNetwork(name string) (*Network, error) {
 		return nil, err
 	}
 	return &Network{
-		network: n.(*object.Network),
+		network: n,
 		driver:  d,
 	}, nil
 }
@@ -37,7 +39,13 @@ func (n *Network) Info(params ...string) (*mo.Network, error) {
 		p = params
 	}
 	var info mo.Network
-	err := n.network.Properties(n.driver.ctx, n.network.Reference(), p, &info)
+
+	network, ok := n.network.(*object.Network)
+	if !ok {
+		return nil, fmt.Errorf("unexpected %t network object type", n.network)
+	}
+
+	err := network.Properties(n.driver.ctx, network.Reference(), p, &info)
 	if err != nil {
 		return nil, err
 	}

